pynxtools/dataconverter/template.py

```diff
@@ -125,6 +125,29 @@ def __contains__(self, k):
             k in self.required
         ])
 
+    def get(self, key, return_value=None):
+        """Implementing get method for template.
+
+        Parameters
+        ----------
+        key : str
+            Template key
+        return_value : Any
+
+        return :
+            The value comes with return_value
+        """
+        val = self.optional.get(key, None)
+        if val is None:
+            val = self.recommended.get(key, None)
+            if val is None:
+                val = self.required.get(key, None)
+                if val is None:
+                    val = self.undocumented.get(key, None)
+        if val is None:
+            return return_value
+        return val
+
```

Comment:
This can just be a proxy to __getitem__. There is a clearer logic of accessing the Template items there. Plus, we don't want duplicate code.

Reply:
I do not know, am I missing something. Usually, `__getintem__` magic method is used for `items()` method, But it does not give method `get()` in dict object. isn't it?

Reply:
```suggestion
    def get(self, key, return_value=None):
        """Implementing get method for template.

        Parameters
        ----------
        key : str
            Template key
        return_value : Any

        return :
            The value comes with return_value
        """
        return self.__getitem__(key)
```

Reply:
> 

@sherjeelshabih I am going back to the old code, the method `self.__getitem__(key)` does not work, because the last statement returns entire require dictionary.

Reply:
Which part returns the whole require dictionary? I don't want duplicate code that is supposed to do the same thing. Let's fix it if the original code is broken. But not duplicate it again.